### DISCUSSION TITLE
How can fetch return immediately when the subject is empty?

### DISCUSSION THREAD

**Discussion by hoslo** [nats-io/nats.go#1460] [Q&A]
### Observed behavior

When i fetch msg on a empty subject, it will wait the maxWait time

### Expected behavior

when subject is empty, fetch return immediately

### Server and client version

server： 2.10.4
client:   github.com/nats-io/nats.go v1.31.0

### Host environment

nats server run on kubernetes
client run another k8s namespace

**Comment by jnmoyne** [comment]
That is the expected behavior. 
To do what you want you can use FetchNoWait (https://github.com/nats-io/nats.go/blob/915b9b8d082f76161d17a47b51c656a4a0462c95/jetstream/consumer.go#L40C14-L40C14) or you can use a very small max wait). 

To check if there are any messages available you can also get the ConsumerInfo for that consumer and check the number of pending messages.
